omegaconf/omegaconf.py

```diff
@@ -825,12 +825,12 @@ def _create_impl(  # noqa F811
             if isinstance(obj, str):
                 obj = yaml.load(obj, Loader=get_yaml_loader())
                 if obj is None:
-                    return OmegaConf.create({}, flags=flags)
+                    return OmegaConf.create({}, parent=parent, flags=flags)
                 elif isinstance(obj, str):
-                    return OmegaConf.create({obj: None}, flags=flags)
+                    return OmegaConf.create({obj: None}, parent=parent, flags=flags)
                 else:
                     assert isinstance(obj, (list, dict))
-                    return OmegaConf.create(obj, flags=flags)
+                    return OmegaConf.create(obj, parent=parent, flags=flags)
```

Comment:
You fixed three code paths:
None, yaml string, and dict/list.

Reply:
They're all from Yaml strings:
* None comes from empty string
* The second case is for a string like "hello"
* Dict/list are the "normal" cases

See the tests I added: they're all based on strings.